pkg/yang/types_builtin.go

```diff
@@ -563,10 +566,11 @@ func FromFloat(f float64) Number {
 	return n
 }
 
-// ParseNumber returns s as a Number.  Numbers may be represented in decimal,
+// ParseInt returns s as a Number with FractionDigits=0.
 // octal, or hexadecimal using the standard prefix notations (e.g., 0 and 0x)
-func ParseNumber(s string) (n Number, err error) {
```

Comment:
Can we not keep the public API here -- since `ParseNumber` returned `Number` then it seems like it should be compatible with both? Is the concern that we can't infer what the `fracDigitRequired` argument should be?

Reply:
Yes exactly, `12.34` -- is that a valid decimal or should it fail? We don't know unless we know that the fractional digit is >= 2 and < 18: https://tools.ietf.org/html/rfc7950#section-9.3.4